packages/fxa-settings/src/pages/Signin/SigninRecoveryChoice/container.test.tsx

```diff
@@ -0,0 +1,147 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as ModelsModule from '../../../models';
+import * as ReachRouterModule from '@reach/router';
+import * as ReactUtils from 'fxa-react/lib/utils';
+import * as CacheModule from '../../../lib/cache';
+import * as SigninRecoveryChoiceModule from './index';
+
+import { waitFor } from '@testing-library/react';
+import { LocationProvider } from '@reach/router';
+import { renderWithLocalizationProvider } from 'fxa-react/lib/test-utils/localizationProvider';
+import { MOCK_STORED_ACCOUNT, mockLoadingSpinnerModule } from '../../mocks';
+import { mockSigninLocationState } from '../mocks';
+import { SigninRecoveryChoiceProps } from '.';
+import SigninRecoveryChoiceContainer from './container';
+import AuthClient from 'fxa-auth-client/lib/client';
+
+jest.mock('../../../models', () => {
+  return {
+    ...jest.requireActual('../../../models'),
+    useAuthClient: jest.fn(),
+  };
+});
+
+const mockAuthClient = new AuthClient('http://localhost:9000', {
+  keyStretchVersion: 1,
+});
+
+function mockModelsModule() {
+  mockAuthClient.recoveryKeyExists = jest.fn().mockResolvedValue({
+    hasBackupCodes: true,
+    count: 3,
+  });
+  mockAuthClient.recoveryPhoneGet = jest
+    .fn()
+    .mockResolvedValue({ exists: true, number: '7890' });
+  (ModelsModule.useAuthClient as jest.Mock).mockImplementation(
+    () => mockAuthClient
+  );
+}
+
+let currentSigninRecoveryChoiceProps: SigninRecoveryChoiceProps | undefined;
+function mockSigninRecoveryChoiceModule() {
+  jest
+    .spyOn(SigninRecoveryChoiceModule, 'default')
+    .mockImplementation((props: SigninRecoveryChoiceProps) => {
+      currentSigninRecoveryChoiceProps = props;
```

Comment:
Maybe this is WIP, but why do this? It'd be more conventional to check what 'default' was called with.

Reply:
Definitely a WIP. This pattern is used in multiple test files, but I'd agree that checking what 'default' is called with is cleaner. I'll try that out, thanks!